backend/franky/service/dockerhub/dockerhub.py

```diff
@@ -0,0 +1,172 @@
+from urllib.parse import urljoin
+from typing import Optional, List, Dict
+from datetime import datetime, timedelta
+
+import requests
+
+from model.user import UserData
+from model.project import ProjectData
+from service import Service
+
+
+class DockerHubException(Exception):
+    pass
+
+
+def user_cleaner(username: str) -> str:
+    """
+    Makes username lowercase
+
+    :param username: user login
+    :return: cleaned username
+    """
+    return username.lower()
+
+
+class DockerHub(Service):
+
+    def __init__(self, rest_endpoint='https://hub.docker.com'):
+        self._version = 'v2'
+        self._rest_endpoint = rest_endpoint
+        self._url = '%s/%s' % (self._rest_endpoint, self._version)
+        self._user_url = 'users'
+        self._tag_url = 'tags'
+        self._repository_url = 'repositories'
+        self._tag_indexes = ['os', 'architecture']
+        self._date_format = '%Y-%m-%dT%H:%M:%SZ'
+        self._active_projects_period = timedelta(weeks=2)
+        self._page_size = 25
+
+    def user(self, username: str) -> UserData:
+        login = user_cleaner(username)
+        full_name = self._get_user_full_name(username)
+        tags = self._get_user_tags(username)
+        return UserData(login=login, name=full_name, tags=tags)
+
+    def _get_user_full_name(self, username) -> str:
+        username = user_cleaner(username)
+        request_url = self._api_url('%s/%s' % (self._user_url, username))
+        payload = dict()
```

Comment:
Maybe it will be better to have a default value for `payload` parameter in `self._call` method?

Reply:
Of course! thanks